lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/util/ConnectorExpressionUtils.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.base.util;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.spi.expression.ConnectorExpression;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+import static io.trino.spi.expression.Constant.TRUE;
+
+public final class ConnectorExpressionUtils
+{
+    private ConnectorExpressionUtils() {}
+
+    public static List<ConnectorExpression> extractConjuncts(ConnectorExpression expression)
+    {
+        // TODO: Implement after adding LogicalExpression to ConnectorExpression
```

Comment:
We're probably not going to add LogicalExpression to ConnectorExpressions, but model it as a function call, instead.

Reply:
Rephrased

Reply:
> We're probably not going to add LogicalExpression to ConnectorExpressions, but model it as a function call, instead.

@martint 

Does that means we only need to use the`visitFunctionCall` method in `ConnectorToSqlExpressionTranslator`? Because many expression can be modeled as a FunctionCall.

Also, I'm a little curious about how we model `LogicalExpression` to a FunctionCall.
Terms in `LogicalExpression` has Indefinite length, however in my understanding LogicalFunction for `AND`&`OR` have two parameters.
In my own implementation, I make a nested Call for `LogicalExpression` which have more than 2 terms.
And in the last, in the pushdown string for `a AND b AND c` will look like `((a AND b) AND c)`.
Is there a better solution?

Reply:
It would be modeled as a function call with variable number of arguments (similar to other functions like `concat`, etc)